components/Feature/AddAction/index.js

```diff
@@ -0,0 +1,107 @@
+import React, { useState } from 'react';
+import { DateInput, TextInput, Button } from '../../Form';
+import TextArea from '../../Form/TextArea';
+
+const AddAction = ({ id }) => {
+  const [summary, setActionSummary] = useState('');
+  const [dueDate, setDueDate] = useState({});
+  const [description, setActionDescription] = useState('');
+  const [button, setActionButtonVisibility] = useState({
+    className: 'govuk-button govuk-button--disabled',
+    disabled: 'disabled'
+  });
+
+  const handleActionSummaryChange = async e => {
+    await setActionSummary(e.target.value);
+    await updateButton();
+  };
+
+  const handleDueDateChange = async e => {
+    let currentDate = dueDate;
+    if (e.target.name.includes('day'))
+      currentDate.day = parseInt(e.target.value);
+    if (e.target.name.includes('month'))
+      currentDate.month = parseInt(e.target.value);
+    if (e.target.name.includes('year'))
+      currentDate.year = parseInt(e.target.value);
+    await setDueDate(currentDate);
+    await updateButton();
+  };
+
+  const handleActionDescriptionChange = async e => {
+    await setActionDescription(e.target.value);
+    await updateButton();
+  };
+
+  const updateButton = async () => {
+    if (summary && dueDate.year && dueDate.month && dueDate.day) {
+      setActionButtonVisibility({ className: 'govuk-button', disabled: '' });
+    } else {
+      setActionButtonVisibility({
+        className: 'govuk-button govuk-button--disabled',
+        disabled: 'disabled'
+      });
+    }
+  };
```

Comment:
This could probably be done in the render, rather than as a separate thing that gets stored in state. Then we don't need to remember to call this method after each update.

Reply:
It's so lovely of you to think that I know how to do it

Reply:
Haha, I believe in you. Happy to go through refactoring this to be inside of the render tomorrow if you like? Happy for this to merge as-is though and we can fix it up afterwards.